checker/src/main/java/org/checkerframework/checker/calledmethods/framework/AutoValueSupport.java

```diff
@@ -0,0 +1,451 @@
+package org.checkerframework.checker.calledmethods.framework;
+
+import com.sun.source.tree.NewClassTree;
+import com.sun.tools.javac.code.Symbol;
+import com.sun.tools.javac.code.Types;
+import com.sun.tools.javac.processing.JavacProcessingEnvironment;
+import java.beans.Introspector;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Objects;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import javax.lang.model.element.AnnotationMirror;
+import javax.lang.model.element.Element;
+import javax.lang.model.element.ElementKind;
+import javax.lang.model.element.ExecutableElement;
+import javax.lang.model.element.Modifier;
+import javax.lang.model.element.TypeElement;
+import javax.lang.model.type.DeclaredType;
+import javax.lang.model.type.TypeKind;
+import javax.lang.model.type.TypeMirror;
+import org.checkerframework.checker.calledmethods.CalledMethodsAnnotatedTypeFactory;
+import org.checkerframework.framework.type.AnnotatedTypeMirror;
+import org.checkerframework.framework.type.AnnotatedTypeMirror.AnnotatedExecutableType;
+import org.checkerframework.framework.util.AnnotatedTypes;
+import org.checkerframework.javacutil.AnnotationUtils;
+import org.checkerframework.javacutil.TreeUtils;
+import org.checkerframework.javacutil.TypesUtils;
+
+/**
+ * AutoValue Support for the Called Methods Checker, which adds CalledMethods annotation to the code
+ * generated by AutoValue.
+ */
+public class AutoValueSupport implements FrameworkSupport {
+
+    /** The type factory. */
+    private CalledMethodsAnnotatedTypeFactory atypeFactory;
+
+    /**
+     * Simple constructor.
+     *
+     * @param atypeFactory the typechecker's type factory
+     */
+    public AutoValueSupport(CalledMethodsAnnotatedTypeFactory atypeFactory) {
+        this.atypeFactory = atypeFactory;
+    }
+
+    /**
+     * For frameworks to add any necessary @{@link
```

Comment:
This comment is copied from the superclass.  If so, you might as well leave it off, since it will be inherited automatically.
Better, say what this method does.